src/index.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useState, useRef, useEffect, useCallback } from 'react'
+import { factories, service, models } from 'powerbi-client'
+import { TokenType, Permissions } from 'powerbi-models'
+import { IEmbedConfiguration, Embed } from 'embed'
+
+const powerbi = new service.Service(
+  factories.hpmFactory,
+  factories.wpmpFactory,
+  factories.routerFactory
+)
+
+type EmbedType = 'report' | 'dashboard' | 'tile'
+
+type PowerbiEmbeddedProps = Pick<IEmbedConfiguration,
+'id' |
+'width' |
+'height' |
+'pageName' |
+'embedUrl' |
+'tokenType' |
+'accessToken' |
+'permissions'> & {
+  onEmbedded?: (embed: Embed) => void;
+  embedType: EmbedType;
+  mobile?: boolean;
+  filterPaneEnabled?: boolean;
+  navContentPaneEnabled?: boolean;
+};
+
+export default function PowerbiEmbedded ({ onEmbedded, ...props }: PowerbiEmbeddedProps): JSX.Element {
```

Comment:
I'm not a fan of namespaces. Can you change `JSX.Element` to `React.FunctionComponentElement<PowerbiEmbeddedProps>`
OR better yet, change the name function to an arrow function like so:
`const PowerbiEmbedded: React.FC<PowerbiEmbeddedProps> = ({ onEmbedded, ...props }) => ...`

Reply:
👍